### PR TITLE
Add packing to wikitext dataset

### PR DESCRIPTION
**Pull request by krammnic**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [X] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)
#1788 
Please link to any issues this PR addresses.

#### Changelog
What are the changes made in this PR?
* Add packing to wikitext dataset.

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [X] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [X] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [X] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [X] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

**Comment by krammnic** [comment]
@RdoubleA @joecummings here everything is okay, might be merged

File: torchtune/datasets/_wikitext.py [hunk 1]
```diff
@@ -47,12 +47,29 @@ def wikitext_dataset(
...
-
+    Raises:
+        ValueError: If ``packed=True`` and ``tokenizer.max_seq_len`` is not set.
     Returns:
         Union[TextCompletionDataset, PackedDataset]: the configured :class:`~torchtune.datasets.TextCompletionDataset`
             or :class:`~torchtune.datasets.PackedDataset` if ``packed=True``
     """
 
+    if packed:
+        if tokenizer.max_seq_len is None:
```
Comment (RdoubleA) [inline torchtune/datasets/_wikitext.py:58]: This error is already checked and raised in `text_completion_dataset` so you don't need to check it again here. Should be sufficient to just pass `packed` directly through

**Review by RdoubleA — COMMENTED** [review]
Thanks for this, could you verify in the test plan that you were able to pack wikitext successfully?

File: torchtune/datasets/_wikitext.py (hunk 1 above)
Comment (krammnic) [reply torchtune/datasets/_wikitext.py:58]: Oh I see. Will be fixed soon.

Comment (krammnic) [reply torchtune/datasets/_wikitext.py]: Probably now is fine

Comment (krammnic) [reply torchtune/datasets/_wikitext.py]: Should be fine now

**Comment by joecummings** [comment]
Looks like you might need to merge/rebase on main?

**Comment by krammnic** [comment]
Yeah, let me fix merge conflict

**Pull request closed without merging**